Warning: MSVC runtime library mismatch (LNK4098)

## Description
Windows builds produce warning LNK4098 about conflicting default libraries:

```
LIBCMT.lib(initializers.obj) : warning LNK4098: defaultlib 'msvcrt.lib' conflicts with use of other libs; use /NODEFAULTLIB:library
```

## Root Cause
Mixed MSVC runtime library settings across targets:
- `DefaultCXX.cmake` files set `MultiThreaded$<$<CONFIG:Debug>:Debug>DLL` → `/MD` (dynamic)
- `SingleExecutable.cmake` sets `MultiThreaded$<$<CONFIG:Debug>:Debug>` → `/MT` (static)
- `libultraship/CMakeLists.txt` sets `/MT` (static)

The game CMakeLists.txt files override to static later, but some targets may slip through.

## Files Affected
- `CMake/DefaultCXX.cmake`
- `games/oot/CMake/DefaultCXX.cmake`
- `games/mm/CMake/DefaultCXX.cmake`
- `OTRExporter/CMake/DefaultCXX.cmake`
- `libultraship/cmake/DefaultCXX.cmake`
- `CMake/SingleExecutable.cmake`

## Suggested Fix
Standardize all targets to use static runtime (`/MT`) consistently. Update DefaultCXX.cmake files to remove the "DLL" suffix.

## Priority
Low - builds succeed but this could cause runtime issues.

--- a/CMake/DefaultCXX.cmake
+++ b/CMake/DefaultCXX.cmake
@@ -6,7 +6,7 @@ if(MSVC)
     create_property_reader("DEFAULT_CXX_EXCEPTION_HANDLING")
     create_property_reader("DEFAULT_CXX_DEBUG_INFORMATION_FORMAT")
 
-    set_target_properties("${PROPS_TARGET}" PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+    set_target_properties("${PROPS_TARGET}" PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
     set_config_specific_property("DEFAULT_CXX_EXCEPTION_HANDLING" "/EHsc")
     # Only set debug info for Debug builds - Release without debug info avoids 4GB COFF limit
     # PROPS_CONFIG comes from the CMake invocation context

--- a/games/mm/CMake/DefaultCXX.cmake
+++ b/games/mm/CMake/DefaultCXX.cmake
@@ -6,7 +6,7 @@ if(MSVC)
     create_property_reader("DEFAULT_CXX_EXCEPTION_HANDLING")
     create_property_reader("DEFAULT_CXX_DEBUG_INFORMATION_FORMAT")
 
-    set_target_properties("${PROPS_TARGET}" PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+    set_target_properties("${PROPS_TARGET}" PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
     set_config_specific_property("DEFAULT_CXX_EXCEPTION_HANDLING" "/EHsc")
     # Only set debug info for Debug builds - Release without debug info avoids 4GB COFF limit
     if(PROPS_CONFIG STREQUAL "Debug")

--- a/games/oot/CMake/DefaultCXX.cmake
+++ b/games/oot/CMake/DefaultCXX.cmake
@@ -6,7 +6,7 @@ if(MSVC)
     create_property_reader("DEFAULT_CXX_EXCEPTION_HANDLING")
     create_property_reader("DEFAULT_CXX_DEBUG_INFORMATION_FORMAT")
 
-    set_target_properties("${PROPS_TARGET}" PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+    set_target_properties("${PROPS_TARGET}" PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
     set_config_specific_property("DEFAULT_CXX_EXCEPTION_HANDLING" "/EHsc")
     # Only set debug info for Debug builds - Release without debug info avoids 4GB COFF limit
     if(PROPS_CONFIG STREQUAL "Debug")

